libraries/chain/db_maint.cpp

```diff
@@ -766,6 +766,49 @@ void database::process_bids( const asset_bitasset_data_object& bad )
    _cancel_bids_and_revive_mpa( to_revive, bad );
 }
 
+void database::process_bitassets()
+{
+   if(head_block_time() >= HARDFORK_CORE_518_TIME)
+   {
+      for( const auto& d : get_index_type<asset_bitasset_data_index>().indices() )
+      {
+         modify(d, [this](asset_bitasset_data_object &o)
+         {
+            o.force_settled_volume = 0; // Reset all BitAsset force settlement volumes to zero
+
+            const auto &asset = get(o.asset_id);
+            auto flags = asset.options.flags;
+            if ((flags & witness_fed_asset) || (flags & committee_fed_asset)) // if smartcoin
+            {
+               // check overflow
+               if (std::numeric_limits<uint32_t>::max() - o.options.feed_lifetime_sec > head_block_time().sec_since_epoch())
```

Comment:
I think this is wrong. Two lines below you calculate (head_block_time - feed_lifetime), so what you want to avoid is feed_lifetime > head_block_time.
Also, please move this check into the if two lines above.

(Good thinking to put in the check though.)

Reply:
@pmconrad i think you are right and making sure feed_lifetime < head is enough. changed here https://github.com/bitshares/bitshares-core/pull/832/commits/960bfe91659cfcd0803bc3e20b78448d7a4be316 and moved check to the previous if. thanks.